server/agoraApi.yaml

```diff
@@ -180,7 +180,490 @@ paths:
     get:
       tags:
         - Goals
-      summary: Manage Listing, creating, updating and deleting user goals
+      summary: Retrieves all available goals, provide limit and offset for pagnation, top limit 100.
```

Comment:
I believe the limit is not maximum 100, but rather that the default is 100 if there is no user specification. I know this is modeled after the documentation for tags, but a look at tagService.js suggests the same is true for tags as well. We should discuss either updating this here, or modifying the service layer to enforce a maximum.

Reply:
getAllTags in the tagService and getAllVisibleResources in resourceService both will include a limit if provided, and otherwise will assign limit to 100 as default. getAllVisibleGoals in goalService does not handle this. I think limit and offset should be implemented into the goalService and that we should stick with this Goals summary in the API Docs, but I will bring this question up in @be-engineer-team channel in Discord to get the teams opinion.

Reply:
Yes, all api's should have a default limit and allow the user to send a custom value using the query parameters. The server level default of 100 can change if want to change it, but should not be left open ended. We can also chose to force an upper limit if we want to avoid extremely heavy queries.